### PR TITLE
MNT: Re-rendered with conda-build 3.21.4, conda-smithy 3.16.0, and co…

### PR DESCRIPTION
**Pull request by carterbox**
…nda-forge-pinning 2021.12.18.06.45.12

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by carterbox** [comment]
Looks like the Windows nvcc host compiler version (14.16.27023) is not matching the one provided as in the cxx_compiler package (19.16.27045.0)? The host compiler is supposed to be automatically picked up using the %CXX% environment variable in [this activation script](https://github.com/conda-forge/nvcc-feedstock/blob/master/recipe/windows/nvcc_windows.bat).

File: recipe/bld.bat [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+echo "CUDA path is %CUDA_PATH%"
+echo "CUDA home is %CUDA_HOME%"
+echo "CUDA compiler is %CUDACXX%"
+%CUDACXX% --version
+echo "CUDA host compiler is %CUDAHOSTCXX%"
+echo "CMAKE_ARGS is %CMAKE_ARGS%"
+cmake -G Ninja -DCMAKE_BUILD_TYPE=Release^
+  -DCMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX%^
+  -DCMAKE_CUDA_COMPILER="%CUDACXX%"^
+  -DCMAKE_CUDA_HOST_COMPILER="%CUDAHOSTCXX%"^
```
Comment (jaimergp) [inline recipe/bld.bat:20]: In CMake, if you pass it manually you need to replace the backslashes with forward slashes
```suggestion
  -DCMAKE_CUDA_COMPILER="%CUDACXX:\=/%"^
  -DCMAKE_CUDA_HOST_COMPILER="%CUDAHOSTCXX:\=/%"^
```

**Comment by jaimergp** [comment]
Something's off with the compiler. `cl.exe` is not found by CMake but the activation script does find it 🤷   Maybe something to do with the manually passed variables (but everything should be already set up by the activation script, so we might not need those at all).

**Comment by carterbox** [comment]
Something is wrong with the build environment!? Because `echo "SRC_DIR is %SRC_DIR%"` prints `"SRC_DIR is %SRC_DIR%"` instead of `SRC_DIR is 'D:\\bld\\libastra_1642194596632\\work'`

**Comment by carterbox** [comment]
@conda-forge-admin please rerender

**Comment by carterbox** [comment]
@jaimergp, I'd be happy to spend some time debugging this Windows Image, but I don't know where to find the Docker image so I can run it locally. Perhaps I should be debugging by trying to run the tests for the nvcc-feedstock?

**Comment by jaimergp** [comment]
There is no official Docker image, sadly. All we can do is to set up a VM and install the compilers. Then Miniconda and the build stack. We do need to write some docs about local debugging on Windows...

Everything looks ok in the meta.yaml and build scripts so I wonder where that "cache cannot be loaded" error is coming :/ 

Maybe this one example that uses CMake on Windows can help? https://github.com/conda-forge/openmm-feedstock/blob/master/recipe/build_openmm.bat

**Comment by carterbox** [comment]
Is there a technical reason that prevents us from using Docker for Windows (Why isn't there a conda-forge Windows anvil)?

**Comment by jaimergp** [comment]
Licensing is one reason. Docker images can only run on Windows hosts and they are HUGE. See [here](https://hub.docker.com/_/microsoft-windows-base-os-images). Ideally we could have one that mimics Azure's but it hasn't been explored.

**Comment by jaimergp** [comment]
@conda-forge-admin, please rerender

**Comment by jaimergp** [comment]
So the previous build used `win-64/vs2017_win-64-19.16.27038-h2e3bad8_2.tar.bz2` as the compiler activator, which is now considered broken. I guess this comes from https://github.com/conda-forge/vc-feedstock/issues/44.

Let's see if we can take another one from the same time (build string `_2`).

**Comment by carterbox** [comment]
@jaimergp, I appreciate your effort on this PR, but I think we should simplify the problem by looking upstream at the nvcc-feedstock and seeing if we can get a very simple CMAKE project to compile.

**Comment by carterbox** [comment]
Or rather, this PR can wait until then because I think solving problems with https://github.com/conda-forge/nvcc-feedstock/pull/59 will resolve issues here.

**Comment by jaimergp** [comment]
Maybe not that PR specifically, but distributing the new CUDA packages Nvidia has been preparing for a while. See [here](https://github.com/conda-forge/cudatoolkit-feedstock/issues/62).

**Comment by carterbox** [comment]
@conda-forge-admin, please rerender

**Pull request closed without merging**